flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -139,13 +163,10 @@ public void close() throws IOException {
 	@Override
 	public boolean reachedEnd() throws IOException {
 		try {
-			if (resultSet.isLast()) {
-				close();
-				return true;
-			}
-			return false;
+			return !resultSet.next();
```

Comment:
The problem is that resultSet.isLast() return false when the resultset is empty and that is the safest solution. Just call it once..